src/name/admitriev/jhelper/actions/ArchiveTaskAction.java

```diff
@@ -0,0 +1,186 @@
+                }
+
+                @Override
+                public void afterChildrenVisited(@NotNull VirtualFile file) {
+                    pathToFile.remove(pathToFile.size() - 1);
+                }
+            });
+            if (archivedCount > 0) {
+                LocalFileSystem.getInstance().refresh(true); // to notify Vfs about new files created in archive
+                Notificator.showNotification(
+                        "All tasks in " + selectedFile.getName() + " were successfully archived",
+                        NotificationType.INFORMATION
+                );
+                DeleteTaskAction.selectSomeTaskConfiguration(RunManagerEx.getInstanceEx(project));
+            }
+        }
+
+        private void handleSourceFile(@NotNull VirtualFile file) {
+            RunnerAndConfigurationSettings taskConfigurationAndSettings = findJHelperRCForFile(file, project);
+            if (taskConfigurationAndSettings == null) {
+                return;
+            }
+            TaskConfiguration taskConfiguration = ((TaskConfiguration) taskConfigurationAndSettings.getConfiguration());
+            VirtualFile directory = getFinalArchiveDirectoryOfFile(file);
+            saveCodeToArchive(file, directory, taskConfiguration.getClassName() + "_" + curDate + ".cpp");
+            saveRCToArchive(taskConfiguration, directory, taskConfiguration.getClassName() + "_" + curDate + ".xml");
+            deleteFileWithCode(file);
+            deleteRC(taskConfigurationAndSettings);
+            ++archivedCount;
+        }
+
+        private void deleteRC(RunnerAndConfigurationSettings taskConfigurationAndSettings) {
+            RunManagerEx runManager = RunManagerEx.getInstanceEx(project);
+            runManager.removeConfiguration(taskConfigurationAndSettings);
+        }
+
+        private void deleteFileWithCode(@NotNull VirtualFile file) {
+            try {
+                file.delete(this);
+            } catch (IOException e) {
+                throw new NotificationException("Archiving of the task failed", "Unable to delete sources of the task, caused by " + e.getMessage());
+            }
+        }
+
+        private void saveCodeToArchive(@NotNull VirtualFile fileWithCode, VirtualFile directoryInArchiveForTheTask, String archiveTaskFileName) {
+            try {
+                VfsUtil.copyFile(this, fileWithCode, directoryInArchiveForTheTask, archiveTaskFileName);
+            } catch (IOException e) {
+                throw new NotificationException("Archiving of the task failed", "Archiving of the code of the solution failed, caused by " + e.getMessage());
+            }
+        }
+
+        private void saveRCToArchive(TaskConfiguration taskConfiguration, VirtualFile directoryInArchiveForTheTask, String archiveRCFileName) {
+            try {
+                String rcArchiveFqn = directoryInArchiveForTheTask.getPath() + "/" + archiveRCFileName;
+                Element rc = new Element("RC");
```

Comment:
It looks like here we do something very similar to what IDE does for us when saving configurations(in `.idea/runConfigurations/`). Maybe it's possible and makes sense to reuse it somehow?

Reply:
I'd spell RC fully in the XML

Reply:
Yes, I tried to do this via their API, but it is not obvious how to make the platform "forget" about the RC stored in a separate file before deleting them. They have a handy method for that in platform [com.intellij.execution.impl.RunManagerImpl#removeConfigurations$intellij_platform_execution_impl](https://upsource.jetbrains.com/idea-ce/file/idea-ce-bd682f0397f4277fc7d0a232300697fd6ac7d1aa/platform/execution-impl/src/com/intellij/execution/impl/RunManagerImpl.kt?nav=44342:44598:focused&line=1368&preview=false) but it is private. 

Alternatively, we can get a path to the file where our configuration is currently stored and just copy the file back and forth. But that seems to be worse since the platform may update the place where RCs are stored in the future and that will break restoring.

Do you see any other options to reuse the platform?

I agree about the spelling, I will fix it